src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -122,10 +124,16 @@ private String metricName(MetricName metricName) {
                 .toLowerCase(Locale.ROOT);
     }
 
-    private static Labels labelsFromTags(Map<String, String> tags) {
+    protected static Labels labelsFromTags(Map<String, String> tags) {
         Labels.Builder builder = Labels.builder();
+        Set<String> labelNames = new HashSet<>();
         for (Map.Entry<String, String> label : tags.entrySet()) {
-            builder.label(PrometheusNaming.sanitizeLabelName(label.getKey()), label.getValue());
+            String newLabelName = PrometheusNaming.sanitizeLabelName(label.getKey());
+            if (labelNames.add(newLabelName)) {
+                builder.label(newLabelName, label.getValue());
+            } else {
+                LOG.warn("Ignoring duplicate label key: {} with value: {}", newLabelName, label.getValue());
```

Comment:
If wonder if it's worth passing the metric name so the message helps identify the impacted metric.

Same in `YammerMetricsCollector`.

Reply:
Metric name passed and added to the LOG